ironic-config/ironic-python-agent.ign.j2

```diff
@@ -42,10 +42,17 @@ inspection_dhcp_all_interfaces = True
   },
   {% endif -%}
   "storage": {
-    "files": [{
-      "path": "/etc/ironic-python-agent.conf",
-      "contents": {"source": "data:,{{ ipa_config | urlencode }}"}
-    }]
+    "files": [
+      {
+        "path": "/etc/ironic-python-agent.conf",
+        "contents": {"source": "data:,{{ ipa_config | urlencode }}"}
+      }{% if env.IRONIC_AGENT_PULL_SECRET %},
+      {
+        "path": "/etc/authfile.json",
+        "contents": {"source": "data:,{{ env.IRONIC_AGENT_PULL_SECRET | trim | urlencode }}"}
```

Comment:
Can I just send you a base64 encoded data here, which comes straight from the secret?

Reply:
Ugh. Sorry to bug you again. I think we need to keep the `urlencode`  back. The secret value I'm getting is plain text and not base64 like I expected.